lib/class-wp-rest-url-details-controller.php

```diff
@@ -216,22 +209,27 @@ private function get_remote_url( $url ) {
 	 * Parses the <title> contents from the provided HTML
 	 *
 	 * @param string $html The HTML from the remote website at URL.
-	 * @return string The title tag contents on success; else empty string.
+	 * @return string The title tag contents on success, or an empty string.
 	 */
 	private function get_title( $html ) {
-		preg_match( '|<title[^>]*>(.*?)<\s*/\s*title>|is', $html, $match_title );
+		$pattern = '#<title[^>]*>(.*?)<\s*/\s*title>#is';
+		preg_match( $pattern, $html, $match_title );
 
-		$title = isset( $match_title[1] ) && is_string( $match_title[1] ) ? trim( $match_title[1] ) : '';
+		$title = ! empty( $match_title[1] ) && is_string( $match_title[1] ) ? trim( $match_title[1] ) : '';
 
-		return $title;
+		if ( empty( $title ) ) {
+			return '';
+		}
+
+		return $this->convert_html_entities( $title );
```

Comment:
@getdave Converting entities in the `<title>` (like doing in description) to ensure consistency and readiness for downstream.